server/src/main/scala/com/cloudera/livy/server/WebServer.scala

```diff
@@ -96,11 +99,11 @@ class WebServer(livyConf: LivyConf, var host: String, var port: Int) extends Log
     val connector = server.getConnectors()(0).asInstanceOf[NetworkConnector]
 
     if (host == "0.0.0.0") {
-      host = InetAddress.getLocalHost.getHostAddress
+      host = InetAddress.getLocalHost.getCanonicalHostName
     }
     port = connector.getLocalPort
 
-    info("Starting server on %s" format port)
+    info("Starting server on %s" format (protocol + "://" + host + ":" + port))
```

Comment:
Could you use interpolation instead of this hybrid of concatenation and `String.format`?